ros_ws/src/crazyswarm/scripts/pycrazyswarm/crazyflie.py

```diff
@@ -96,26 +215,104 @@ def uploadTrajectory(self, trajectoryId, pieceOffset, trajectory):
         self.uploadTrajectoryService(trajectoryId, pieceOffset, pieces)
 
     def startTrajectory(self, trajectoryId, timescale = 1.0, reverse = False, relative = True, groupMask = 0):
+        """Begins executing a previously uploaded trajectory.
+
+        Asynchronous command; returns immediately.
+
+        Args:
+            trajectoryId (int): ID number as given to :meth:`uploadTrajectory()`.
+            timescale (float): Scales the trajectory duration by this factor.
+                For example if timescale == 2.0, the trajectory will take twice
+                as long to execute as the nominal duration.
+            reverse (bool): If true, executes the trajectory backwards in time.
+            relative (bool): If true (default), the position of the trajectory
+                is shifted such that it begins at the current position setpoint.
+                This is usually the desired behavior.
+            groupMask (int): Group mask bits. See :meth:`setGroupMask()` doc.
+        """
         self.startTrajectoryService(groupMask, trajectoryId, timescale, reverse, relative)
 
     def position(self):
+        """Returns the true position as determined by motion capture.
+
+        Blocking command. Waits for a fresh update from the mocap system.
+        TODO: make a non-blocking version that potentially gives stale data.
+
+        Returns:
+            position (np.array[3]): Current position. Meters.
+        """
         self.tf.waitForTransform("/world", "/cf" + str(self.id), rospy.Time(0), rospy.Duration(10))
         position, quaternion = self.tf.lookupTransform("/world", "/cf" + str(self.id), rospy.Time(0))
         return np.array(position)
 
     def getParam(self, name):
+        """Returns the current value of the onboard named parameter.
+
+        Parameters are named values of various primitive C types that control
+        the firmware's behavior. For more information, see
+        https://www.bitcraze.io/docs/crazyflie-firmware/master/logparam/.
+
+        Parameters are read at system startup over the radio and cached.
+        The ROS launch file can also be used to set parameter values at startup.
+        Subsequent calls to :meth:`setParam()` will update the cached value.
+        However, if the parameter changes for any other reason (TODO: do any
+        other reasons exist?), the cached value might become stale.
```

Comment:
Bitcraze tried to avoid this case. There are a few exceptions, for example "trigger" parameters that are used to reset the EKF.